demos/projects/ESPRESSIF/az-nvs-cert-bundle/main/az_nvs_cert_bundle_main.c

```diff
@@ -0,0 +1,283 @@
+        return err;
+    }
+
+/* Skip this if user wishes to force write the bundle. */
+    #ifndef AZ_FORCE_WRITE
+        /* If version matches, do not write to not overuse NVS */
+        if( read_trust_bundle_version == trust_bundle_version )
+        {
+            printf( "Trust bundle version in NVS matches bundle version to write.\n" );
+            return ESP_OK;
+        }
+    #endif
+
+    /* Write value including previously saved blob if available */
+    printf( "Writing trust bundle\n" );
+    err = nvs_set_blob( my_handle, AZURE_TRUST_BUNDLE_NAME, trust_bundle, trust_bundle_size );
+
+    if( err != ESP_OK )
+    {
+        printf( "Error (%s) getting AZURE_TRUST_BUNDLE_VERSION_NAME from NVS!\n", esp_err_to_name( err ) );
+        return err;
+    }
+
+    /* Set new trust bundle version */
+    printf( "Writing trust bundle version\n" );
+    err = nvs_set_i32( my_handle, AZURE_TRUST_BUNDLE_VERSION_NAME, trust_bundle_version );
+
+    if( err != ESP_OK )
+    {
+        printf( "Error (%s) getting AZURE_TRUST_BUNDLE_VERSION_NAME from NVS!\n", esp_err_to_name( err ) );
+        return err;
+    }
+
+    /* Commit */
+    err = nvs_commit( my_handle );
+
+    if( err != ESP_OK )
+    {
+        printf( "Error (%s) getting AZURE_TRUST_BUNDLE_VERSION_NAME from NVS!\n", esp_err_to_name( err ) );
+        return err;
+    }
+
+    printf( "Printing what was just written\n" );
+    err = print_saved_bundle( my_handle );
+
+    return err;
+}
+
+/*
+ *  Read and write trust bundle
+ */
+void read_and_write_bundle( void )
+{
+    nvs_handle_t my_handle;
+    esp_err_t err;
+
+    /* Open CA Cert namespace */
+    err = nvs_open( CA_CERT_NAMESPACE, NVS_READWRITE, &my_handle );
+
+    if( err != ESP_OK )
+    {
+        printf( "Error (%s) opening CA_CERT_NAMESPACE in NVS\n", esp_err_to_name( err ) );
+        return;
+    }
+
+    printf( "||| Printing what is currently saved |||\n" );
+
+    err = print_saved_bundle( my_handle );
+
+    if( err != ESP_OK )
+    {
+        printf( "Error (%s) printing saved trust bundle!\n", esp_err_to_name( err ) );
+    }
+    else
+    {
+        printf( "||| Check if bundle needs to be written. Write if needed. |||\n" );
+
+        err = save_trust_bundle( my_handle );
+
+        if( err != ESP_OK )
+        {
+            printf( "Error (%s) saving trust bundle to NVS!\n", esp_err_to_name( err ) );
+            nvs_close( my_handle );
+            return;
+        }
+    }
+
+    nvs_close( my_handle );
+}
+
+void app_main( void )
+{
+    /* Delay to give time to open UART console */
+    vTaskDelay( 5000 / portTICK_PERIOD_MS );
+
+    esp_err_t err = nvs_flash_init();
+
+    if( ( err == ESP_ERR_NVS_NO_FREE_PAGES ) || ( err == ESP_ERR_NVS_NEW_VERSION_FOUND ) )
+    {
+        /* NVS partition was truncated and needs to be erased */
+        /* Retry nvs_flash_init */
+        ESP_ERROR_CHECK( nvs_flash_erase() );
+        err = nvs_flash_init();
```

Comment:
The `ESP_ERROR_CHECK()` should be sufficient since this code block handles the attempt to rectify the error. If there's any error at this point, it should be checked with it.